sdk/storage/azure-storage-blob/azure/storage/blob/aio/_blob_client_async.py

```diff
@@ -460,6 +460,28 @@ async def undelete_blob(self, **kwargs):
         except StorageErrorException as error:
             process_storage_error(error)
 
+    @distributed_trace_async
+    async def exists(self, **kwargs):
+        # type: (**Any) -> bool
+        """
+        Returns True if a blob exists with the defined parameters, and returns
+        False otherwise.
+
+        :param str version_id:
+            The version id parameter is an opaque DateTime
+            value that, when present, specifies the version of the blob to check if it exists.
+        :param int timeout:
+            The timeout parameter is expressed in seconds.
+        :returns: boolean
+        """
+        try:
+            await self._client.blob.get_properties(
+                snapshot=self.snapshot,
+                **kwargs)
+            return True
+        except StorageErrorException:
+            return False
```

Comment:
```suggestion
        except StorageErrorException as error:
            try:
                process_storage_error(error)
            except ResourceNotFoundError:
                return False
```

Reply:
If it's authentication error or other error, we don't want to tell the user the the blob doesn't exist, we should throw instead

Reply:
When a blob doesn't exist the response error is a `StorageErrorException`, the `ResourceNotFoundError` doesn't get thrown for some odd reason

Reply:
process_storage_error will parse the exception and ResourceNotFoundError will be thrown from there